### PR TITLE
Calculate height of section levels

### PR DESCRIPTION
**Pull request by mkeshavarzi**
Calculates z value for a section plane for each level + a camera location based on IFC model height and a screen size Coefficient. Please note main.test.js and flags.test.js have been temp removed to allow push. Need to resolve.

### CODE REVIEW

**Comment by pablo-mayrgundter** [comment]
Heya, I'm happy to review.  While you still have tests excluded please leave in draft.  I just flipped it.

Are there sample screens?

**Comment by mkeshavarzi** [comment]
> Heya, I'm happy to review. While you still have tests excluded please leave in draft. I just flipped it.
> Are there sample screens?

Thanks! No graphics in this one yet, just calculating coordinates.

**Comment by pablo-mayrgundter** [comment]
Got it checked out with ```git checkout upstream/CalculateSectionAndCamera```, though leaves the local client in "detached" state.  Anyways..

I uncomment calSecLevels and run:
```
> node lib/src/calLevSec.js ~/bldrs/3d\ models/IFC/Schependomlaan.ifc out.ifc
Input file:  /Users/pablo/bldrs/3d models/IFC/Schependomlaan.ifc
web-ifc: 0.0.34 threading: 0
```

But no output to out.ifc.

**Comment by mkeshavarzi** [comment]
> Got it checked out with `git checkout upstream/CalculateSectionAndCamera`, though leaves the local client in "detached" state. Anyways..
> 
> I uncomment calSecLevels and run:
> 
> ```
> > node lib/src/calLevSec.js ~/bldrs/3d\ models/IFC/Schependomlaan.ifc out.ifc
> Input file:  /Users/pablo/bldrs/3d models/IFC/Schependomlaan.ifc
> web-ifc: 0.0.34 threading: 0
> ```
> 
> But no output to out.ifc.

The script doesn't output an ifc. It outputs a simple array of height values. 

To see the values, please uncomment line 31:
```console.log(elevValuesAll)```

and run:
```node lib/src/calLevSec.js ifcFileName```
(in your case):
```node lib/src/calLevSec.js ~/bldrs/3d\ models/IFC/Schependomlaan.ifc```

for example:
```mohammad@CheetahM1 tools % node lib/src/calLevSec.js lib/src/testdata/haus.ifc                
Input file:  lib/src/testdata/haus.ifc
web-ifc: 0.0.34 threading: 0
[ 0, 2.7 ]```

**Comment by pablo-mayrgundter** [comment]
Hi Mohammed.  Sorry, I lost track.  Everything's running locally for me, though you have the existing tests deleted.  Just remove them from this PR.

What were you blocked on for testing?

**Comment by mkeshavarzi** [comment]
> Hi Mohammed. Sorry, I lost track. Everything's running locally for me, though you have the existing tests deleted. Just remove them from this PR.
> 
> What were you blocked on for testing?

My block is I can't run main.test.js and flags.test.js, and hence cannot push the code without removing them.

**Comment by pablo-mayrgundter** [comment]
Ah, I see.  Not sure how to fix this.  The version of ifclib that's being referenced is somehow being confused.  The main branch is ahead of your branch and has fixed those reference problems.

I tried updating tool/package.json to depend on the upstream version, 5.3.3, and then doing yarn install, but that doesn't seem to fix it:

```
> grep 5.3 tool/package.json
  "version": "5.3.1",
    "@bldrs-ai/ifclib": "^5.3.3"
```
```
> yarn install
> npm ls
@bldrs-ai/ifctool-root@5.3.1 /Users/pablo/c/pm/ifctool
├── @babel/core@7.18.6
├── @babel/eslint-parser@7.18.2
├── @babel/plugin-syntax-import-assertions@7.18.6
├── @babel/preset-env@7.18.6
├─┬ @bldrs-ai/ifclib@5.3.1 -> ./lib
│ ├── @log4js-node/log4js-api@1.0.2
│ ├── json2csv@5.0.7
│ └── web-ifc@0.0.35
├─┬ @bldrs-ai/ifctool@5.3.1 -> ./tool
│ ├── @bldrs-ai/ifclib@5.3.3.                    # GOOD!
│ └── log4js@6.6.0
├── esbuild@0.14.48
├── eslint-config-google@0.14.0
├── eslint-plugin-import@2.26.0
├── eslint@8.19.0
├── husky@7.0.4
├── jest@27.5.1
├── shx@0.3.4
└── three@0.142.0
```
However:
```
> grep version node_modules/\@bldrs-ai/ifclib/package.json
  "version": "5.3.1"
```
and indeed tests still fail with the bad ref to Exception which was indeed missing from from 5.3.1

File: tool/src/main.js [hunk 1]
```diff
@@ -142,4 +142,4 @@ export async function processArgs(args, print=console.log) {
 }
 
 
-process.exitCode = await processArgs(process.argv.slice(2))
+process.exitCode = await processArgs(process.argv.slice(2))
```
Comment (pablo-mayrgundter) [inline tool/src/main.js:145]: pls restore newline at eof

File: tool/src/main.test.js [hunk 2]
```diff
@@ -60,4 +60,4 @@ describe('main', () => {
     expect(consoleLogMock).toBeCalledTimes(1)
     consoleLogMock.mockRestore()
   })
-})
+})
```
Comment (pablo-mayrgundter) [inline tool/src/main.test.js:63]: newline at eof

File: tool/src/flags.js [hunk 3]
```diff
@@ -10,7 +10,8 @@ export function parseFlags(args) {
   for (let i = 0; i < args.length; i++) {
     let flag = args[i]
     if (!flag.startsWith('--')) {
-      throw new Exception('Trailing arguments must be in format: --flag or --flag=value')
+      // throw new Exception('Trailing arguments must be in format: --flag or --flag=value')
+      throw new Error
```
Comment (pablo-mayrgundter) [inline tool/src/flags.js:14]: pls revert

**Review by pablo-mayrgundter — CHANGES_REQUESTED** [review]

**Pull request merged**